### PR TITLE
upgrade: Save the backup location info in the progress file

### PR DESCRIPTION
**Pull request by jsuchome**
Requires https://github.com/crowbar/crowbar-core/pull/1011

### CODE REVIEW

**Comment by jsuchome** [comment]
[(3) Implement API for providing backup of OpenStack database during upgrade (part of upgrade step 9-10)](https://trello.com/c/DFaqwV10/116-3-implement-api-for-providing-backup-of-openstack-database-during-upgrade-part-of-upgrade-step-9-10)

File: crowbar_framework/lib/crowbar/upgrade_status.rb [hunk 1]
```diff
@@ -55,7 +55,10 @@ def initialize_state
         current_node: nil,
         # number of nodes still to be upgraded
         remaining_nodes: nil,
-        upgraded_nodes: nil
+        upgraded_nodes: nil,
+        # locations of the backups took during the upgrade
```
Comment (skazi0) [inline crowbar_framework/lib/crowbar/upgrade_status.rb]: taken?

File: crowbar_framework/lib/crowbar/upgrade_status.rb [hunk 2]
```diff
@@ -149,6 +152,20 @@ def cancel_allowed?
       ].include?(current_step) && !running?(:admin)
     end
 
+    def save_crowbar_backup(backup_location = "")
```
Comment (MaximilianMeister) [inline crowbar_framework/lib/crowbar/upgrade_status.rb]: i'd not allow an empty backup_location here, or put some reasonable default

Comment (jsuchome) [inline crowbar_framework/lib/crowbar/upgrade_status.rb]: Well, `reasonable default` here is strange. Library itself should not know about that path if if it wasn't provided with the call, there's no reason to guess...

Comment (jsuchome) [inline crowbar_framework/lib/crowbar/upgrade_status.rb]: And if it is empty string, that's nothing wrong from the library POV.

Comment (MaximilianMeister) [inline crowbar_framework/lib/crowbar/upgrade_status.rb]: it's just when you don't submit any parameter the path will be empty, isn't that wrong? why not just enforce the path by not giving a default?

Comment (skazi0) [inline crowbar_framework/lib/crowbar/upgrade_status.rb]: +1 on not giving the default. This is a clear indication to the caller that something needs to be passed.

**Comment by jsuchome** [comment]
@MaximilianMeister @skazi0 removed default value

**Pull request merged**